### ISSUE TITLE
System with no users in group "wheel" is very vulnerable

### ISSUE THREAD

**Issue by jussihi** [linuxmint/cinnamon#7594]
```
 * Cinnamon version 3.8.3
 * Distribution - Arch
 * 64 bit
 ```

**Issue**
I have a setup with client user with automatic login (no password) and with very restricted permissions. Issue: the user may change/remove other users' details/passwords from cinnamon's Users and Groups app without logging in as the admin account. The system doesn't have users in group "wheel".

**Steps to reproduce**
Remove all users from "wheel" group. Create a user with no password and with auto login. Open "Users and Groups" and grab admin rights.

**Expected behaviour**
I would like to see some sort of password prompt for root or some user account with sudo rights (well as I told I don't have wheel group in sudoers, but let's say group "staff"). I don't think this account with very low privileges should be able to administrate all the users.

**Other information**
I'm very sorry if this is wrong place to make an issue about it, but I really couldn't find anything useful from internet.

**Comment by jussihi** [comment]
Update:
This "bug" only happens if no users belong to the "wheel" group on the system. Is it intended to work like this?

**Comment by eli-schwartz** [comment]
The "Users and Groups" dialog is run using pkexec, the policy for which can be seen in /usr/share/polkit-1/actions/org.cinnamon.settings-users.policy

So the question is not, why does cinnamon let you do this, and more "what did you do to your polkit configuration to make auth_admin not ask for a password"?

**Comment by jussihi** [comment]
@eli-schwartz I did not change anything inside that polkit configuration file. But I guess it assumes that at least one user account belongs to the group "wheel". If not, it will assume that everyone has admin rights. Or at least it seems like so. In my opinion there should be a warning popup for this kind of systems with no users in group "wheel".

**Comment by icarter09** [comment]
Is this a Cinnamon-specific issue?

**Comment by jussihi** [comment]
@icarter09 , I did not test this on other DEs. It might be that it's some sort of PolKit backup if no users belong to "wheel".

**Comment by claudiux** [comment]
I think it's not an issue. A system on which the root user does not have a password (which is the default case in Linux Mint), it seems logical that the first user can declare himself as an administrator (it is that is, as part of the wheel group), if no administrator already exists.

**Comment by eli-schwartz** [comment]
Right, again, cinnamon is just asking for auth_admin, and it is polkit that decided you could immediately be eligible for auth_admin without a password dialog.

See the Arch wiki documentation for details on configuring polkit: https://wiki.archlinux.org/index.php/Polkit

Note: you've claimed that the user you were logged in as was *not* in the wheel group. Arch Linux comes with a default rule which adds the "wheel" group as administrative identities, therefore allowing you to select "wheel" group members in the polkit authentication dialog and type in the password of said group members to "log in" as that wheel user and do admin things.

Did you get any selection dialog to choose a user?

Is it possible you've created or installed some rule which bypasses the password prompt as per https://wiki.archlinux.org/index.php/Polkit#Bypass_password_prompt ?

**Comment by jussihi** [comment]
@eli-schwartz , sorry, it's been over a year I encountered this issue. I cannot remember for sure, but afaik it just showed the Users and Groups page as if I had administrative rights to the system (I could edit everything). No dialog was shown if I remember correctly...

I did not edit any polkit rules, but it might be that I installed some software that did. I'm sorry, I don't have the configuration anymore so I cannot tell.

All in all, I got it working after complying with the Arch default PolKit rules and adding a user to the "wheel" group.
